backend/services/product/internal/product/product.go

```diff
@@ -0,0 +1,116 @@
+package product
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/manaaan/ekolivs-oms/pkg/env"
+	"github.com/manaaan/ekolivs-oms/pkg/zettle"
+	"github.com/manaaan/ekolivs-oms/product/api"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type Service struct {
+	zettleService *zettle.Service
+	storeService  *Store
+}
+
+func New() (*Service, error) {
+	zettleService, err := zettle.New(zettle.ServiceNewParams{
+		ClientId: env.Required("ZETTLE_ORG_UUID"),
+		ApiKey:   env.Required("ZETTLE_API_KEY"),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Service{
+		zettleService: zettleService,
+		storeService:  &Store{},
+	}, nil
+}
+
+func (s Service) GetProducts() ([]*api.Product, error) {
+	zettleProducts, err := s.zettleService.GetProducts()
+	if err != nil {
+		return nil, err
+	}
+
+	storeProducts, err := s.storeService.GetProducts()
+	fmt.Println(storeProducts)
+
+	products := []*api.Product{}
+	for _, zettleProduct := range zettleProducts {
+		for _, variant := range zettleProduct.Variants {
+			var imageUrl *string
+			if zettleProduct.Presentation != nil {
+				imageUrl = zettleProduct.Presentation.ImageUrl
+			}
+
+			products = append(products, &api.Product{
+				Id:            variant.Uuid.String(),
+				Name:          zettleProduct.Name,
```

Comment:
We cannot simply use the name of the product, but need to build a combination with the variant name as well.

Example: 

```
"id": "abd56de0-2edf-11ee-b919-43f578ecb5ae",
"name": "Skånsk Chili (250ml) - BS",
```

```
"id": "abd56de1-2edf-11ee-b919-43f578ecb5ae",
"name": "Skånsk Chili (250ml) - BS",
```

ID is with a slight change on the first part unique.

```diff
- abd56de0
+ abd56de1
```

Reply:
Good point. For now I recommend concatenating them with a hyphen to keep it simple: `Skånsk Chili (250ml) - BS - Rostad Habanero`. What do you think?